### ISSUE TITLE
SimpleCopy Paste Augmentation Implementation

### ISSUE THREAD

**Issue by sudz123** [open-mmlab/mmdetection#6191]
**Simple Copy Paste is a data augmentation technique which is extremely useful for generating data for training and testing.**

**Motivation**
There is a recent paper https://arxiv.org/abs/2012.07177,  which is very helpful for improving scores in instance segmentation and classification.

**Related resources**
https://github.com/tensorflow/tpu/tree/master/models/official/detection/projects/copy_paste

**Additional context**
Hi, we have implemented this feature already in mmdet for the LVIS 2021 challenge and we have made our submission. We want to add our code to the official repository and hence submit a PR. We feel this would be a great addition to the repository. 

If we get the go ahead, me, along with @GrantorShadow and @akki2503 will work on formatting our code, adding tests and so on, so that we can cleanly add it to the repo.

**Comment by hhaAndroid** [comment]
@sudz123 thank you very much. If you have any questions, please feel free to contact us.

**Comment by ZwwWayne** [comment]
Hi @sudz123 ,
PRs are definitely welcomed! We will also organize the PR review as soon as the PRs are created.
To make the progress more smooth, it is also preferred if you could briefly describe your plan, e.g., the module you want to add and the related potential modification. You can also describe that following the PR template when creating the PR.

**Comment by sudz123** [comment]
Hi @ZwwWayne Can you point out where we can find the PR template?

To summarise the plan:

1. We will add one new Class to ```mmdet.datasets.pipelines.transforms``` called ```SimpleCopyPaste```. The code will exactly match the [official implementation](https://github.com/tensorflow/tpu/tree/master/models/official/detection/projects/copy_paste).
2. This class will take in two images at random and hence use the class ```MultiImageMixDataset``` from mmdet.datasets.dataset_wrappers.py 
3. We will provide one simple config which uses ```mask rcnn``` with ```SimpleCopyPaste```. 
4. We can also provide trained model results for a small backbone like ResNet18, ResNet50 and put this in the ReadMe in the configs.

We, however, cannot give the exact backbone implementation given in the original paper and the corresponding model since it was trained using a large batch size for very long and we do not have the compute power to replicate it. I hope that is okay?

We will submit the PR within this week and based on comments, we will modify and then close it as soon as possible.

**Comment by AronLin** [comment]
> Hi @ZwwWayne Can you point out where we can find the PR template?
> 
> To summarise the plan:
> 
> 1. We will add one new Class to `mmdet.datasets.pipelines.transforms` called `SimpleCopyPaste`. The code will exactly match the [official implementation](https://github.com/tensorflow/tpu/tree/master/models/official/detection/projects/copy_paste).
> 2. This class will take in two images at random and hence use the class `MultiImageMixDataset` from mmdet.datasets.dataset_wrappers.py
> 3. We will provide one simple config which uses `mask rcnn` with `SimpleCopyPaste`.
> 4. We can also provide trained model results for a small backbone like `ResNet18`, `ResNet50` and put this in the ReadMe in the configs.
> 
> We, however, cannot give the exact backbone implementation given in the original paper and the corresponding model since it was trained using a large batch size for very long and we do not have the compute power to replicate it. I hope that is okay?
> 
> We will submit the PR within this week and based on comments, we will modify and then close it as soon as possible.

When you creating a PR, the template will be shown on the web page.

**Comment by TalentedMUSE** [comment]
Hello, I am interested in your inplementation and I wonder if SimpleCopy Paste is available. Also, how can I add this one to my project (for example, configs and the place of the [SimpleCopy Paste] original code). Thank you~~

**Comment by ZwwWayne** [comment]
Already in https://github.com/open-mmlab/mmdetection/tree/master/configs/simple_copy_paste through https://github.com/open-mmlab/mmdetection/pull/7501
